crmsh/utils.py

```diff
@@ -2512,34 +2473,125 @@ def check_file_content_included(source_file, target_file):
     return source_data in target_data
 
 
-def service_is_enabled(service, remote_addr=None):
+class ServiceManager(object):
     """
-    Check if service is enabled
+    Class to manage systemctl services
     """
-    cmd = "systemctl is-enabled {}".format(service)
-    if remote_addr:
-        # check on remote
-        prompt_msg = "Check whether {} is enabled on {}".format(service, remote_addr)
-        rc, _, _ = run_cmd_on_remote(cmd, remote_addr, prompt_msg)
-    else:
-        # check on local
-        rc, _ = get_stdout(cmd)
-    return rc == 0
+    ACTION_LIST = ["enable", "disable", "start", "stop", "is-enabled", "is-active", "list-unit-files"]
 
+    def __init__(self, service_name, remote_addr=None):
+        """
+        Init function
+        """
+        self.service_name = service_name
+        self.remote_addr = remote_addr
 
-def service_is_active(service, remote_addr=None):
-    """
-    Check if service is active
-    """
-    cmd = "systemctl -q is-active {}".format(service)
-    if remote_addr:
-        # check on remote
-        prompt_msg = "Check whether {} is active on {}".format(service, remote_addr)
-        rc, _, _ = run_cmd_on_remote(cmd, remote_addr, prompt_msg)
-    else:
-        # check on local
-        rc, _ = get_stdout(cmd)
-    return rc == 0
+    def _do_action(self, action_type):
+        """
+        Actual do actions to manage service
+        """
+        if action_type not in self.ACTION_LIST:
+            raise ValueError("status_type should be {}".format('/'.join(self.ACTION_LIST)))
+
+        cmd = "systemctl {} {}".format(action_type, self.service_name)
+        if self.remote_addr:
+            prompt_msg = "Run \"{}\" on {}".format(cmd, self.remote_addr)
+            rc, output, err = run_cmd_on_remote(cmd, self.remote_addr, prompt_msg)
+        else:
+            rc, output, err = get_stdout_stderr(cmd)
+        if rc != 0 and err:
+            raise ValueError("Run \"{}\" error: {}".format(cmd, err))
+        return rc == 0, output
+
+    def is_available(self):
+        return self.service_name in self._do_action(self.ACTION_LIST[-1])[1]
```

Comment:
This looks pretty ugly (`self.ACTION_LIST[-1]`). I would change the `ACTION_LIST` from being a list to be a map or named list. The code would be neater.